src/runtime/HalideRuntime.h

```diff
@@ -513,40 +542,18 @@ extern int halide_device_malloc(void *user_context, struct halide_buffer_t *buf,
 /** Free device memory. */
 extern int halide_device_free(void *user_context, struct halide_buffer_t *buf);
 
-/** Get a pointer to halide_device_free if a Halide runtime has been
- * linked in. Returns null if it has not. This requires a different
- * mechanism on different platforms. */
-typedef int (*halide_device_free_t)(void *, struct halide_buffer_t *);
-#ifdef _MSC_VER
-extern const __declspec(selectany) void *halide_dummy_device_free = NULL;
-extern int halide_weak_device_free(void *user_context, struct halide_buffer_t *buf);
-// The following pragma tells the windows linker to make
-// halide_device_free_weak the same symbol as halide_dummy_device_free
-// if it can't resolve halide_weak_device_free normally
-#ifdef _WIN64
-#pragma comment(linker, "/alternatename:halide_weak_device_free=halide_dummy_device_free")
-#else
-#pragma comment(linker, "/alternatename:_halide_weak_device_free=_halide_dummy_device_free")
-#endif
-inline halide_device_free_t halide_get_device_free_fn() {
-    if ((const void **)(&halide_weak_device_free) == &halide_dummy_device_free) {
-        return NULL;
-    } else {
-        return &halide_weak_device_free;
-    }
-};
-#elif __MINGW32__
-inline halide_device_free_t halide_get_device_free_fn() {
-    // There is no workable mechanism for doing this that we know of on mingw.
-    return &halide_device_free;
-}
-#else
-extern __attribute__((weak)) int halide_weak_device_free(void *user_context, struct halide_buffer_t *buf);
-inline halide_device_free_t halide_get_device_free_fn() {
-    return &halide_weak_device_free;
-}
-#endif
-
+/** Wrap or detach a native device handle. The meaning of the opaque
+ * is specific to the device interface, so if you know the device
+ * interface in use, call the more specific functions in the runtime
+ * headers for your specific device API instead
+ * (e.g. HalideRuntimeCuda.h). */
+// @{
+extern int halide_device_wrap_native(void *user_context,
```

Comment:
No, there are no such requirements. It just sets the device field to the texture id. I'll clarify in the comment.